tests/pyright/decorators.py

```diff
@@ -88,6 +90,58 @@ def valid_method_no_info(self) -> Self: ...
     def valid_method_info(self, info: ValidationInfo) -> Self: ...
 
 
+class BeforeFieldValidator(BaseModel):
+    """Same tests should apply to `mode='plain'`."""
+
+    @field_validator('foo', mode='before')
+    def no_classmethod(self, value: Any) -> Any:
+        """TODO this shouldn't be valid, the decorator should only work on classmethods.
+
+        We might want to do the same type checking as wrap model validators.
+        """
+
+    @field_validator('foo', mode='before')
+    @classmethod
+    def valid_classmethod(cls, value: Any) -> Any: ...
+
+    @field_validator('foo', mode='before')  # pyright: ignore[reportArgumentType]
+    @classmethod
+    def invalid_with_info(cls, value: Any, info: int) -> Any: ...
+
+    @field_validator('foo', mode='before', json_schema_input_type=int)  # `input_type` allowed here.
```

Comment:
Let's modify the comment to `json_schema_input_type` allowed here, and add the same one for the below `WrapFieldValidator` case